Mathlib/Topology/ContinuousFunction/ContinuousMapZero.lean

```diff
@@ -66,37 +70,64 @@ instance instPartialOrder [PartialOrder R] : PartialOrder C(X, R)₀ :=
 
 lemma le_def [PartialOrder R] (f g : C(X, R)₀) : f ≤ g ↔ ∀ x, f x ≤ g x := Iff.rfl
 
+protected instance instTopologicalSpace : TopologicalSpace C(X, R)₀ :=
+  TopologicalSpace.induced toContinuousMap inferInstance
+
+lemma embedding_toContinuousMap : Embedding (toContinuousMap (X := X) (R := R)) where
+  induced := rfl
+  inj _ _ h := ext fun x ↦ congr($(h) x)
+
+instance [T0Space R] : T0Space C(X, R)₀ := embedding_toContinuousMap.t0Space
+instance [T1Space R] : T1Space C(X, R)₀ := embedding_toContinuousMap.t1Space
+instance [T2Space R] : T2Space C(X, R)₀ := embedding_toContinuousMap.t2Space
+
+lemma closedEmbedding_toContinuousMap [T1Space R] :
```

Comment:
Here, it would be enough to assume that `{0 : R}` is closed.

Reply:
I feel this is a case where the loss of generality is acceptable for the following reasons:
- the version you describe would be slightly more annoying to use because we wouldn't have the typeclass system to infer the arguments in the obvious cases.
- I don't expect the more general version to be useful. First because there's not a bunch of interesting examples where the distinction would matter (e.g in a topological group the two conditions are equivalent), second because this type is really an ad-hoc definition meant for one specific application (the more general setup would be pointed continuous maps where you allow more freedom on the point).
- Finally, if someone ever needs this generalization it would be very easy to add.

Does that make sense?

Reply:
It does, especially because the `R` is likely to be `Real`.